src/workerd/api/node/async-hooks.h

```diff
@@ -29,12 +29,25 @@ namespace workerd::api::node {
 //   console.log(als.getStore());  // undefined
 class AsyncLocalStorage final: public jsg::Object {
  public:
-  AsyncLocalStorage(): key(kj::refcounted<jsg::AsyncContextFrame::StorageKey>()) {}
+  struct AsyncLocalStorageOptions {
+    jsg::Optional<jsg::JsRef<jsg::JsValue>> defaultValue;
+    jsg::Optional<kj::String> name;
+    JSG_STRUCT(defaultValue, name)
+  };
+
+  AsyncLocalStorage(jsg::Optional<AsyncLocalStorageOptions> options = kj::none)
+      : key(kj::refcounted<jsg::AsyncContextFrame::StorageKey>()) {
+    KJ_IF_SOME(opt, options) {
+      defaultValue = kj::mv(opt.defaultValue), name = kj::mv(opt.name);
```

Comment:
Nit: I don't like using the comma operator in cases like this but that is just my personal take.

```suggestion
      defaultValue = kj::mv(opt.defaultValue);
      name = kj::mv(opt.name);
```

Reply:
huh, I don't remember actually typing it that way. Wonder if the formatter changed it. Either way, yep, I agree with the suggestion!